backend/dataall/tasks/data_sharing/share_managers/s3_share_manager.py

```diff
@@ -286,61 +303,77 @@ def delete_access_point_policy(self):
                 access_point_policy["Statement"].remove(statements[f"{target_requester_id}1"])
         S3.attach_access_point_policy(self.source_account_id, self.source_environment.region, self.access_point_name, json.dumps(access_point_policy))
 
-    def delete_access_point(self):
+    @staticmethod
+    def delete_access_point(
+            share: models.ShareObject,
+            dataset: models.Dataset,
+    ):
+        access_point_name = S3ShareManager.build_access_point_name(share)
         logger.info(
-            f'Deleting access point {self.access_point_name}...'
+            f'Deleting access point {access_point_name}...'
         )
-        access_point_policy = json.loads(S3.get_access_point_policy(self.source_account_id, self.source_environment.region, self.access_point_name))
+        access_point_policy = json.loads(S3.get_access_point_policy(dataset.AwsAccountId, dataset.region, access_point_name))
         if len(access_point_policy["Statement"]) <= 1:
             # At least we have the 'AllowAllToAdmin' statement
-            S3.delete_bucket_access_point(self.source_account_id, self.source_environment.region, self.access_point_name)
+            S3.delete_bucket_access_point(dataset.AwsAccountId, dataset.region, access_point_name)
             return True
         else:
             return False
 
-    def delete_target_role_access_policy(self):
+    @staticmethod
+    def delete_target_role_access_policy(
+            share: models.ShareObject,
+            dataset: models.Dataset,
+            target_environment: models.Environment,
+    ):
         logger.info(
             'Deleting target role IAM policy...'
         )
+        access_point_name = S3ShareManager.build_access_point_name(share)
         existing_policy = IAM.get_role_policy(
-            self.target_account_id,
-            self.target_requester_IAMRoleName,
+            dataset.AwsAccountId,
```

Comment:
Here you we want to delete the policy on the requester account, therefore we should use the target_environments AwsAccountId as input for the IAM.get_role_policy() fct.


`        existing_policy = IAM.get_role_policy(
            target_environment.AwsAccountId,
            share.principalIAMRoleName,
            "targetDatasetAccessControlPolicy",
        )`